### ISSUE TITLE
Test that package groups are actually installable

### ISSUE THREAD

**Issue by alexiri** [CentOS/sig-core-t_functional#68]
Currently, CS8 is unable to install a package group because of some broken dependencies in qt5 packages:

```
# dnf group install platform-devel
CentOS Stream 8 - CERN                                                                                       96 kB/s |  24 kB     00:00    
CentOS Stream 8 - AppStream                                                                                  14 MB/s | 7.5 MB     00:00    
CentOS Stream 8 - AppStream - testing                                                                        11 MB/s | 7.5 MB     00:00    
CentOS Stream 8 - BaseOS                                                                                    5.6 MB/s | 2.6 MB     00:00    
CentOS Stream 8 - Extras                                                                                     42 kB/s | 9.3 kB     00:00    
CentOS Stream 8 - PowerTools                                                                                8.1 MB/s | 2.2 MB     00:00    
Error:
 Problem: package qt5-qtcanvas3d-5.12.5-2.el8.x86_64 requires qt5-qtbase(x86-64) = 5.12.5, but none of the providers can be installed
  - cannot install both qt5-qtbase-5.15.2-2.el8.x86_64 and qt5-qtbase-5.12.5-8.el8.x86_64
  - package qt5-qtxmlpatterns-devel-5.15.2-1.el8.x86_64 requires libQt5Core.so.5(Qt_5.15)(64bit), but none of the providers can be installed
  - cannot install both qt5-qtbase-5.12.5-8.el8.x86_64 and qt5-qtbase-5.15.2-2.el8.x86_64
  - cannot install the best candidate for the job
(try to add '--allowerasing' to command line to replace conflicting packages or '--skip-broken' to skip uninstallable packages or '--nobest' to use not only best candidate packages)
```

I'd like to develop a test to catch these sorts of errors before they're pushed to the mirrors. As it might take too long to install all the package groups in a reproducible manner during the tests, I was trying to think of ways of testing this without actually installing the packages. I was thinking of using `dnf --debugsolver` and grepping the output of `debugdata/rpms/solver.result` for "problem".

Is there a better way to see that all packages are actually installable? Or should I just install in an installroot anyway?

**Comment by alexiri** [comment]
[This](https://github.com/CentOS/sig-core-t_functional/compare/master...alexiri:comps) would be the implementation I described, please let me know what you think and I'll open a PR.

**Comment by arrfab** [comment]
haven't looked at that particular issue but that's what previous repoclosure test was supposed to find, but due to issue with dnf on el8, in fact repoclosure (the tool) doesn't work at all  ... so yeah, finding another way to find same result would be good (never looked at option)

**Comment by alexiri** [comment]
I totally missed the repoclosure test! I see it does fail if I force it to run with `PRE_UPDATES=1 ./runtests.sh z_repoclosure`, but it lists some unresolved dependencies that do in fact look ok.

I can replace the repoclosure test with my approach, if you want. Should I move it to the `z_*` namespace? Should I also add the `if [ $SKIP_QA_HARNESS -eq 1 ] && [ "$PRE_UPDATES" != "1" ]  then skip` bit? I'm not entirely sure under what conditions those variables are set.

@carlwgeorge I'm assuming the repoclosure test is disabled in your pipelies?

**Comment by carlwgeorge** [comment]
I'm not familiar with the PRE_UPDATES variable, but looking at a recent run for CS8 it does appear that z_repoclosure is skipped.

```
20:51:49 [+] Tue  4 May 01:51:49 UTC 2021 -> Running ./tests/z_repoclosure/repoclosure_test.sh -  repoclosure test .
20:51:49 [+] Tue  4 May 01:51:49 UTC 2021 -> Skip this test in non QA harness environment
```

The repoclosure test is still applicable for C7, so don't replace it.  I think naming it test_comps works fine.  No sense in naming it repoclosure if it doesn't use the repoclosure tool.

**Comment by alexiri** [comment]
@arrfab do you have any more feedback on #69?

**Comment by reenberg** [comment]
Since the issue is not closed, my I suggest that this test is also done after the Epel repos have been added?  I just found a dependency error in Stream 9, after enabling the Epel repo and installing the Workstation environment group, as Epel pulled in the `rhc` package which suddenly started having a hard dependency on `insights-client`.

As I understand, this test would have caught that.

Ref: https://bugzilla.redhat.com/show_bug.cgi?id=2064944
